cli/tests/test_checkout.rs

```diff
@@ -100,6 +100,40 @@ fn test_checkout_not_single_rev() {
     "###);
 }
 
+#[test]
+fn test_checkout_conflicting_branches() {
+    // Much of this test is borrowed from `test_git_fetch_conflicting_branches` in
+    // test_git_fetch.rs
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "one"]);
+    test_env.jj_cmd_success(&repo_path, &["new", "-m", "two", "@-"]);
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "foo"]);
+    test_env.jj_cmd_success(&repo_path, &["--at-op=@-", "branch", "create", "foo", "-r", r#"description("one")"#]);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["log"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Concurrent modification detected, resolving automatically.
+    @  kkmpptxz test.user@example.com 2001-02-03 04:05:09.000 +07:00 foo?? 66c6502d
+    │  (empty) two
+    │ ◉  qpvuntsm test.user@example.com 2001-02-03 04:05:08.000 +07:00 foo?? a9330854
+    ├─╯  (empty) one
+    ◉  zzzzzzzz root() 00000000
+    "###);
```

Comment:
I actually made this as a hack since `jj_cmd_failure` below asserts that `stdout` is empty and `"Concurrent modification detected, resolving automatically."` ends up coming as stdout  😅. I can think of something better thought.

Reply:
Ah, right. I didn't notice the "Concurrent modification detected, resolving automatically.". I'd be fine with this:

```rust
// Trigger resolution of concurrent operations
jj_cmd_success(&["st"]);
```